book/workflow.rst

```diff
@@ -0,0 +1,236 @@
+
+* `Workflows`_ are a way to apply states and transitions that are triggered by user actions through
+  the UI.
+
+* If you want to automate some steps, you can use `Processes`_ which are performed in the
+  background.
+
+Workflows
+---------
+
+Workflows are a way to apply states to entities and to control when and how an entity can perform
+the transition from one state to another.
+
+.. seealso::
+
+    Simple workflows can be created in the user interface. Read more about this feature
+    :doc:`in the user guide </user_guide/workflow_management>`.
+
+.. sidebar:: Workflows, Steps, and Transitions
+
+    The state of an entity is called a *step*. It is represented by an instance of the
+    :class:`Oro\\Bundle\\WorkflowBundle\\Entity\\WorkflowStep` class. The process of moving an
+    entity from one step to another is called a *transition*.
+
+Configuring a Workflow
+~~~~~~~~~~~~~~~~~~~~~~
+
+Custom workflows can be configured using a YAML configuration format that will be explained below.
+This configuration needs to be placed in a file named ``workflow.yml`` in the ``Resources/config``
+directory of your bundle.
+
+General information
+...................
+
+Under the ``workflows`` key each workflow is created by adding a key that is its name and is used
+to identify workflows. The following example configures a workflow that defines the process of
+calling a customer:
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/Acme/DemoBundle/Resources/config/workflow.yml
+    workflows:
+        phone_call:
+            label: 'Call Customer'
+            entity: Acme\Bundle\DemoBundle\Entity\PhoneCall
+            enabled: true
+            start_step: start_call
+
+The basic configuration contains a name that is displayed in the UI, the entity the workflow will
+be applied to and the initial step that is applied when initializing the workflow for an entity.
+
+Defining the Workflow Steps
+...........................
+
+Next, you need to define the states an entity can have when the workflow is applied:
+
+.. code-block:: yaml
+    :linenos:
+
+    workflows:
+        phone_call:
+            # ...
+            steps:
+                start_call:
+                    label: 'Start Phone Call'
+                    allowed_transitions:
+                        - connected
+                        - not_answered
+                start_conversation:
+                    label: 'Call Phone Conversation'
+                    allowed_transitions:
+                        - end_conversation
+                end_call:
+                    label: 'End Phone Call'
+                    is_final: true
+
+In this example, the entity can have three states: ``start_call`` (which was configured as the
+start step above), ``start_conversation`` and ``end_call``. The ``end_call`` step is marked as
+final which means that the workflow terminates when this step is reached.
```

Comment:
In fact, "final" flag on a step does not restrict any workflow manipulations - you can do transitions from this step etc. "Final" flag is used as informational mark of a step for some additional data manipulations - like building of report for entities that have workflow in final step.

Real example - Sales Funnel Flow (https://github.com/orocrm/crm/blob/5ed72e9e3ba63be3eb128917391cfa92c9c0312a/src/OroCRM/Bundle/SalesBundle/Resources/config/oro/workflow/b2b_flow_sales_funnel.yml). This flow has two final steps - "Won Opportunity" and "Lost Opportunity" - but after each of them customer still can do reopen and start whole workflow basically from scratch.